### ISSUE TITLE
el sistema esta en español pero los paises estan en ingles

### ISSUE THREAD

**Issue by alanboy** [omegaup/omegaup#1222]
Ademas de que no están en orden alfabético

![image](https://cloud.githubusercontent.com/assets/527987/26479512/aef875d0-4188-11e7-88d6-5216ce33239a.png)

**Comment by heduenas** [comment]
Countries show up in sorted order as per https://github.com/omegaup/omegaup/pull/1861 but they still show up in English even when user language is set to Spanish, we need to fix that.

**Comment by Divyansh00** [comment]
In the database, only the English name of countries are stored, so to show spanish names for the country we have to add spanish name for each country

**Comment by NdubuisiO** [comment]
This link seems to have name in Spanish can you confirm  @heduenas http://www.101languages.net/spanish/country-names-spanish/

**Comment by heduenas** [comment]
Yes, those are Spanish names. I think we will also need the 2 letter codes and Portuguese names.

**Comment by Divyansh00** [comment]
@heduenas if we are using the same country codes, we can store the country name in all languages(English, Spanish and Portuguese)  separated by some delimiter(like comma) in the database and then extract the required country language by converting them to string by using split function.
This may not be optimal solution but will do the work.

**Comment by heduenas** [comment]
I think a cleaner solution would be to store the translations in the .lang files as wordsMX, wordsJP, wordsUS, etc. and let the existing localization tools do the localization for us. @lhchavez, do you think that's a good approach?

**Comment by SpaceWhite** [comment]
@heduenas Do you think #1861 is working. In the docs it states that the return of `sort` function is either true or false. So I think it is still not working http://php.net/manual/en/function.sort.php#refsect1-function.sort-returnvalues. And it seems like this issue has not been assigned to anyone else. Can someone assign me this issue?

**Comment by heduenas** [comment]
@SpaceWhite What I understand from the link that you shared is that sort() does sort the array and additionally "returns TRUE on success or FALSE on failure". So, from that perspective, I don't see issues with #1861. However, we still need to fix the localization part of this issue.

**Comment by SpaceWhite** [comment]
@heduenas But the for looping is using that sort return value (true/false) as looping array. I think this would be help https://ideone.com/AZZsJz.

**Comment by heduenas** [comment]
You are right the code is assuming the return value (true/false) of the sort() function is the sorted data itself. However, I just noticed that the sort function in #1861 seems to be being used smarty template file, that means it's not php. I'm not sure how that sort() function works in smarty however. The best thing to do I think would be to just test it out, the change should be in sandbox.omegaup.com already.

**Comment by heduenas** [comment]
X_X the list of countries seems to be empty in the sandbox. @lhchavez sandbox has the list of countries populated in the DB, right? If so, we need to get this fixed (or at least reverted) quick.

**Comment by carlosabcs** [comment]
@tvanessa tal vez esto debas asignarlo en algunos de los próximos milestones, pienso que no es difícil, solo laborioso.

**Comment by lhchavez** [comment]
_muy_ laborioso. no encontramos una lista canónica con los nombres en español y en portugués.
